src/lib/snark_keys/gen_keys/gen_keys.ml

```diff
@@ -172,7 +172,7 @@ let location_expr key_location =
                 key_location)])]
 
 let gen_keys () =
-  let%bind tx_keys_location, tx_keys, tx_keys_checksum =
+  let%bind (tx_keys_location, tx_keys, tx_keys_checksum), r1 =
```

Comment:
Nit: can you rename r1 and r2 to something else? I think it would be hard to read in the future without knowing the context of this PR.